core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/GraalVM.java

```diff
@@ -109,10 +116,23 @@ private static String graalVersion(String buildInfo) {
                 return null;
             }
             String version = buildInfo.substring(idx + JVMCI_BUILD_PREFIX.length());
-            return matchVersion(version);
+            Matcher versMatcher = VERSION_PATTERN.matcher(version);
+            if (versMatcher.find()) {
+                return matchVersion(version);
+            } else {
+                return GRAAL_MAPPING.get(jdkFeature);
+            }
         }
     }
 
+    // Temporarily work around https://github.com/quarkusio/quarkus/issues/36246,
+    // till we have a consensus on how to move forward in
+    // https://github.com/quarkusio/quarkus/issues/34161
```

Comment:
OK, I suppose we can live with this workaround for a short time but let's make progress on this issue please and decide something viable for the future.